lang/cs/Org.Apache.REEF.Bridge.CLR/Network.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Net;
+using System.Threading;
+using org.apache.reef.bridge.message;
+using Org.Apache.REEF.Common.Files;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.Local;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Wake.Remote.Impl;
+
+namespace Org.Apache.REEF.Bridge
+{
+    /// <summary>
+    /// Avro message protocol network to send messages to the java bridge.
+    /// </summary>
+    public sealed class Network
```

Comment:
We use packages to segregate functionality. In this case the name of the class is really Org.Apache.REEF.Bridge.CLR.Network (which will end up once the current C++ bridge goes away Org.Apache.REEF.Bridge.Network). What I would like to avoid is names like Org.Apache.REEF.Bridge.CLR.BridgeNetwork as we already know it is associated with the Bridge due to the package name. I will give the name some thought and respond.